baseline.gradle

```diff
@@ -74,8 +65,7 @@ subprojects {
     spotless {
       java {
         target 'src/main/java/**/*.java', 'src/test/java/**/*.java', 'src/testFixtures/java/**/*.java', 'src/jmh/java/**/*.java', 'src/integration/java/**/*.java'
-        // we use an older version of google-java-format that is compatible with JDK 8
-        googleJavaFormat("1.7")
+        googleJavaFormat("1.23.0")
```

Comment:
I was going back and forth fixing format issues. `1.19.2` was released with `spotless-plugin-gradle:6.25.0`. The error was weirdly reported on non-existent codes.

```
 You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
      src/main/java/org/apache/iceberg/transforms/Transforms.java
          @@ -67,7 +67,9 @@

           ····return·new·UnknownTransform<>(transform);
For more on this, please refer to https://docs.gradle.org/8.10.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
           ··}
25 actionable tasks: 25 executed
           
          -··/**·@deprecated·use·{@link·#identity()}·instead;·will·be·removed·in·2.0.0·*/
          +··/**
          +···*·@deprecated·use·{@link·#identity()}·instead;·will·be·removed·in·2.0.0
          +···*/
           ··@Deprecated
           ··public·static·Transform<?,·?>·fromString(Type·type,·String·transform)·{
           ····Matcher·widthMatcher·=·HAS_WIDTH.matcher(transform);
  Run './gradlew :iceberg-api:spotlessApply' to fix these violations.
```

Reply:
@manuzhang the issue is that spotless will pick a different gjf version depending on the JDK it is executed with. Therefore you'll need to pin the gjf to the latest version that is supported by JDK 11 to produce consistent formatting results across JDK 11/17/21

Reply:
@manuzhang I did a quick experiment in https://github.com/apache/iceberg/pull/11110 and we'll have to stick with `1.17.0`. please make sure to re-introduce the comment from L77 so that it's clear why we're using this particular gjf version